packages/hdwallet-provider/src/index.ts

```diff
@@ -271,21 +276,36 @@ class HDWalletProvider {
         // @ts-ignore - the type doesn't take into account the possibility
         // that response.error could be a thing
         (error: any, response: JSONRPCResponsePayload & { error?: any }) => {
+          // always resolve the Promise - if an error occurs, throw it when
+          // this.initialized is checked so errors can be handled by the user
           if (error) {
-            reject(error);
+            resolve({
+              success: false,
+              error
+            });
             return;
           } else if (response.error) {
-            reject(response.error);
+            resolve({
+              success: false,
+              error: response.error
+            });
             return;
           }
           if (isNaN(parseInt(response.result, 16))) {
             const message =
               "When requesting the chain id from the node, it" +
               `returned the malformed result ${response.result}.`;
-            throw new Error(message);
+            const error = new Error(message);
+            resolve({
+              success: false,
+              error
+            });
```

Comment:
Should there be a return in this branch? it would seem if `isNaN` is true, then resolve will be invoked twice, the second time setting `this.chainId` to NaN

Reply:
IIRC, the second resolve will have an actual impact, but it's still weird to rely on that behavior.

Reply:
Yep I'll add the `return`s.